test/refactoring/ConvertAsync/convert_to_continuation.swift

```diff
@@ -0,0 +1,351 @@
+// SHADOW-CONT-NAME-NEXT:     withoutAsyncAlternativeBecauseOfMismatchedCompletionHandlerName { continuation1 in
+// SHADOW-CONT-NAME-NEXT:       continuation.resume(returning: continuation1)
+// SHADOW-CONT-NAME-NEXT:     }
+// SHADOW-CONT-NAME-NEXT:   }
+// SHADOW-CONT-NAME-NEXT: }
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=SHADOW-CONT-NAME-2 %s
+func testShadowContName2(completionHandler: (Int) -> Void) {
+  let continuation = 3
+  withoutAsyncAlternativeBecauseOfMismatchedCompletionHandlerName { result in
+    completionHandler(result + continuation)
+  }
+}
+// SHADOW-CONT-NAME-2:      func testShadowContName2() async -> Int {
+// SHADOW-CONT-NAME-2-NEXT:   let continuation = 3
+// SHADOW-CONT-NAME-2-NEXT:   return await withCheckedContinuation { (continuation1: CheckedContinuation<Int, Never>) in 
+// SHADOW-CONT-NAME-2-NEXT:     withoutAsyncAlternativeBecauseOfMismatchedCompletionHandlerName { result in
+// SHADOW-CONT-NAME-2-NEXT:       continuation1.resume(returning: result + continuation)
+// SHADOW-CONT-NAME-2-NEXT:     }
+// SHADOW-CONT-NAME-2-NEXT:   }
+// SHADOW-CONT-NAME-2-NEXT: }
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=VOID-RETURN %s
+func testVoidReturnValue(completionHandler: () -> Void) {
+  asyncVoidWithoutAlternative {
+    completionHandler()
+  }
+}
+// VOID-RETURN:      func testVoidReturnValue() async {
+// VOID-RETURN-NEXT:   return await withCheckedContinuation { (continuation: CheckedContinuation<Void, Never>) in 
+// VOID-RETURN-NEXT:     asyncVoidWithoutAlternative {
+// VOID-RETURN-NEXT:       continuation.resume(returning: ())
+// VOID-RETURN-NEXT:     }
+// VOID-RETURN-NEXT:   }
+// VOID-RETURN-NEXT: }
+
+
+
+// Reduced version of https://twitter.com/peterfriese/status/1397835146133479428
+
+class DataTask {
+  let completionHandler: (String?, Error?) -> Void
+
+  init(completionHandler: @escaping (String?, Error?) -> Void) {
+    self.completionHandler = completionHandler
+  }
+  func resume() {
+    completionHandler("mock result", nil)
+  }
+}
+
+class URLSession {
+  static let shared = URLSession()
+
+  func dataTask(completionHandler: @escaping (String?, Error?) -> Void) -> DataTask {
+    return DataTask(completionHandler: completionHandler)
+  }
+}
+
+func processURLResult(_ data: String) throws -> Int {
+  return data.count
+}
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=URL-SESSION %s
+func testDataTask(_ completion: @escaping (Int?) -> Void) {
+  let dataTask = URLSession.shared.dataTask { (data, error) in
+    guard let data = data else {
+      return // Yes, there is a bug about not calling completion here, but that's copied from the Twitter link above
```

Comment:
At least we have checked continuations I guess 😨. Ends up being a log, which is better than nothing...

Reply:
I mean, it was broken before. There’s not much more we can do 🤷‍♂️

Reply:
Yeah, arguably the log is better than what it was 😆